python/lsst/coadd/utils/coaddDataIdContainer.py

```diff
@@ -72,3 +72,44 @@ class ExistingCoaddDataIdContainer(CoaddDataIdContainer):
     def makeDataRefList(self, namespace):
         super(ExistingCoaddDataIdContainer, self).makeDataRefList(namespace)
         self.refList = [ref for ref in self.refList if ref.datasetExists()]
+
+class TractDataIdContainer(CoaddDataIdContainer):
+    def makeDataRefList(self, namespace):
+        """Make self.refList from self.idList
+        It's difficult to make a data reference that merely points to an entire
+        tract: there is no data product solely at the tract level.  Instead, we
+        generate a list of data references for patches within the tract.
+        """
+        datasetType = namespace.config.coaddName + "Coadd"
+        validKeys = set(["tract", "filter", "patch",])
+
+        getPatchRefList = lambda tract: [namespace.butler.dataRef(datasetType=datasetType, tract=tract.getId(),
+                                                                  filter=dataId["filter"],
+                                                                  patch="%d,%d" % patch.getIndex()) for
+                                         patch in tract]
+
+        tractRefs = {} # Data references for each tract
+        for dataId in self.idList:
+            for key in validKeys:
+                if key in ("tract", "patch",):
+                    # Will deal with these explicitly
+                    continue
+                if key not in dataId:
+                    raise argparse.ArgumentError(None, "--id must include " + key)
+
+            skymap = self.getSkymap(namespace)
+
+            if "tract" in dataId:
+                tractId = dataId["tract"]
+                tractRefs = defaultdict(list)
```

Comment:
You probably want to remove this line from the loop, and instead use it as a replacement for line 91.